demos/fluid/main.cpp

```diff
@@ -0,0 +1,142 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2021 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  Description:
+ *    Application description for Fluid Simulation demo.
+ *
+ **************************************************************************/
+
+#include <cinder/CameraUi.h>
+#include <cinder/app/App.h>
+#include <cinder/app/RendererGl.h>
+#include <cinder/gl/gl.h>
+
+#include <CL/sycl.hpp>
+
+#include "fluid.h"
+
+// Size of the fluid container edge (always square shaped).
+const int SIZE{300};
+
+// Default scale at which fluid container is rendered.
+const int SCALE{3};
+
+class FluidSimulationApp : public ci::app::App {
+ public:
+  FluidSimulationApp() : size_{SIZE}, fluid_{SIZE, 0.2f, 0.0f, 0.0000001f} {
+    // Set window size.
+    getWindow()->setSize({size_ * SCALE, size_ * SCALE});
+    // Add usage instruction to application title
+    getWindow()->setTitle(
+        "Fluid Simulation - Move mouse to add fluid - Press space to clear "
+        "fluid");
+  }
+
+  // Create fluid texture.
+  void setup() override {
+    texture_ = ci::gl::Texture2d::create(nullptr, GL_RGBA, size_, size_,
+                                         ci::gl::Texture2d::Format()
+                                             .dataType(GL_UNSIGNED_BYTE)
+                                             .internalFormat(GL_RGBA));
+  }
+
+  // Called once per frame to update the fluid.
+  void update() override {
```

Comment:
I think it would make sense to mark these functions, or even the entire class, as `final`

Reply:
Done